lib/providers/sensors_provider.dart

```diff
@@ -0,0 +1,269 @@
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:flutter/material.dart';
+import 'package:geolocator/geolocator.dart';
+import 'package:sensors_plus/sensors_plus.dart';
+
+class SensorsProvider extends ChangeNotifier{
+  late List<_PositionData> _currentPosition;
+  late List<_AccelerometerData> _accelerometerEvents;
+  late List<_UserAccelerometerData> _userAccelerometerEvents;
+  late List<_MagnetometerData> _magnetometerEvents;
+  late List<_GyroscopeData> _gyroscopeEvents;
+
+  static const Duration _ignoreDuration = Duration(milliseconds: 20);
+  
+  bool _run = false;
+
+  DateTime? _userAccelerometerUpdateTime;
+  DateTime? _accelerometerUpdateTime;
+  DateTime? _gyroscopeUpdateTime;
+  DateTime? _magnetometerUpdateTime;
+  DateTime? _currentPositionUpdateTime;
+
+  int? _userAccelerometerLastInterval;
```

Comment:
let's avoid nullable variables as much as possible and use a default value, specially with primitives like int